instructor/client_genai.py

```diff
@@ -32,9 +32,13 @@ def from_genai(
     use_async: bool = False,
     **kwargs: Any,
 ) -> instructor.Instructor | instructor.AsyncInstructor:
-    assert mode in {
-        instructor.Mode.GENAI_TOOLS,
-    }, "Mode must be one of {instructor.Mode.GENAI_TOOLS}"
+    assert (
+        mode
+        in {
+            instructor.Mode.GENAI_TOOLS,
```

Comment:
can this be replaced with TOOLS and STRUCTURED_OUTPUTS and detect the providers themselves?
that way if I change I don't have to change the tool, or will we have reasonable defaults that it won't matter

Reply:
Hmm I'd rather have a specialised mode like we do for GEMIN_TOOLS. It makes it easier to handle it down the line in the processing of the kwargs and the re-asking because it's an explicit enum here